core/src/Kokkos_View.hpp

```diff
@@ -19,13 +19,1023 @@
+          ((dst_mdspan::rank_dynamic() >= 6) ||
+           (dst.static_extent(5) == src.extent(5))) &&
+          ((dst_mdspan::rank_dynamic() >= 7) ||
+           (dst.static_extent(6) == src.extent(6))) &&
+          ((dst_mdspan::rank_dynamic() == 8) ||
+           (dst.static_extent(7) == src.extent(7))));
+}
+
+namespace Impl {
+template <class... Properties>
+struct BasicViewFromTraits {
+  using view_traits        = ViewTraits<Properties...>;
+  using mdspan_view_traits = MDSpanViewTraits<view_traits>;
+  using element_type       = typename view_traits::value_type;
+  using extents_type       = typename mdspan_view_traits::extents_type;
+  using layout_type        = typename mdspan_view_traits::mdspan_layout_type;
+  using accessor_type      = typename mdspan_view_traits::accessor_type;
+
+  using type =
+      BasicView<element_type, extents_type, layout_type, accessor_type>;
+};
+}  // namespace Impl
+
+template <class DataType, class... Properties>
+struct ViewTraits;
+
+} /* namespace Kokkos */
+
+namespace Kokkos {
+
+template <class DataType, class... Properties>
+class View;
+
+template <class>
+struct is_view : public std::false_type {};
+
+template <class D, class... P>
+struct is_view<View<D, P...> > : public std::true_type {};
+
+template <class D, class... P>
+struct is_view<const View<D, P...> > : public std::true_type {};
+
+template <class T>
+inline constexpr bool is_view_v = is_view<T>::value;
+
+template <class DataType, class... Properties>
+class View : public Impl::BasicViewFromTraits<DataType, Properties...>::type {
+ private:
+  template <class, class...>
+  friend class View;
+  template <typename V>
+  friend struct Kokkos::Impl::ViewTracker;
+
+  using base_t =
+      typename Impl::BasicViewFromTraits<DataType, Properties...>::type;
+
+ public:
+  using base_t::base_t;
+
+  // typedefs originally from ViewTraits
+  using traits               = ViewTraits<DataType, Properties...>;
+  using value_type           = typename traits::value_type;
+  using const_value_type     = typename traits::const_value_type;
+  using non_const_value_type = typename traits::non_const_value_type;
+  using data_type            = DataType;
+  using const_data_type      = typename traits::const_data_type;
+  using non_const_data_type  = typename traits::non_const_data_type;
+  using view_tracker_type    = Impl::ViewTracker<View>;
+  using array_layout         = typename traits::array_layout;
+  using device_type          = typename traits::device_type;
+  using execution_space      = typename traits::execution_space;
+  using memory_space         = typename traits::memory_space;
+  using pointer_type         = typename traits::value_type*;
+  using memory_traits        = typename traits::memory_traits;
+  using host_mirror_space    = typename traits::host_mirror_space;
+  using index_type           = typename traits::memory_space::size_type;
+
+  // FIXME: Should be unsigned
+  using size_type = typename memory_space::size_type;
```

Comment:
Is this a future deprecation? I don't like that `size_type` is different from BasicView's `size_type`

Reply:
We keep it but make it consistently unsigned. This should also likely be
```c++
using size_type = typename traits::size_type;
```